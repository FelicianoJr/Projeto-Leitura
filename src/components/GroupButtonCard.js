import React from "react";
import PropTypes from "prop-types";

const GroupButtonCard = ({
  showComment,
  newComment,
  edit,
  remove,
  voteUp,
  voteDown,
  data
}) => {
  const buttonsShow = data.commentCount === undefined;
  return (
    <div>
      {!buttonsShow && (
        <button
          id="show"
          type="button"
          className="btn btn-outline-dark btn-sm mr-sm-2"
          onClick={showComment}
        >
          Visualisar Comentários{" "}
          <span className="badge badge-light">{data.commentCount}</span>
        </button>
      )}

      {!buttonsShow && (
        <button
          id="edit"
          type="button"
          className="btn btn-outline-dark btn-sm mr-sm-2"
          onClick={newComment}
        >
          Adicionar Comentário
        </button>
      )}
      <button
        id="edit"
        type="button"
        className="btn btn-outline-dark btn-sm mr-sm-2"
        onClick={edit}
      >
        Editar
      </button>
      <button
        id="delete"
        type="button"
        className="btn btn-outline-dark btn-sm mr-sm-2"
        onClick={remove}
      >
        Excluir
      </button>
      <button
        id="like"
        type="button"
        className="btn btn-outline-success btn-sm mr-sm-2"
        onClick={voteUp}
      >
        Gostei
      </button>
      <button
        id="dislike"
        type="button"
        className="btn btn-outline-danger btn-sm"
        onClick={voteDown}
      >
        Não Gostei
      </button>
    </div>
  );
};

GroupButtonCard.protoTypes = {
  showComment: PropTypes.func,
  newComment: PropTypes.func,
  edit: PropTypes.func,
  remove: PropTypes.func,
  voteUp: PropTypes.func,
  voteDown: PropTypes.func
};

export default GroupButtonCard;
