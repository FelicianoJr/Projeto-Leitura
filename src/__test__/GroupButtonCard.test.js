import React from "react";
import { shallow } from "enzyme";
import GroupButtonCard from "../components/GroupButtonCard";

const setup = data => {
  const action = {
    showComment: jest.fn(),
    newComment: jest.fn(),
    edit: jest.fn(),
    remove: jest.fn(),
    voteUp: jest.fn(),
    voteDown: jest.fn()
  };

  const component = shallow(<GroupButtonCard data={data} {...action} />);

  return {
    component: component,
    action,
    action,
    button: component.find("button")
  };
};

describe("<GroupButtonCard>", () => {
  it("should simulate click showComment", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(0).type()).toEqual("button");
    button.at(0).simulate("click");
    expect(action.showComment).toBeCalled();
  });

  it("should simulate click newComment", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(1).type()).toEqual("button");
    button.at(1).simulate("click");
    expect(action.newComment).toBeCalled();
  });

  it("should simulate click edit", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(2).type()).toEqual("button");
    button.at(2).simulate("click");
    expect(action.edit).toBeCalled();
  });

  it("should simulate click remove", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(3).type()).toEqual("button");
    button.at(3).simulate("click");
    expect(action.remove).toBeCalled();
  });

  it("should simulate click voteUp", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(4).type()).toEqual("button");
    button.at(4).simulate("click");
    expect(action.voteUp).toBeCalled();
  });

  it("should simulate click voteDown", () => {
    const { button, action } = setup({ commentCount: 2 });
    expect(button.at(5).type()).toEqual("button");
    button.at(5).simulate("click");
    expect(action.voteDown).toBeCalled();
  });

  
});
