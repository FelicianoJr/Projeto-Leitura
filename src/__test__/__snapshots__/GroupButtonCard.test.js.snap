// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GroupButtonCard> should component Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroupButtonCard
    data={
        Object {
            "commentCount": 2,
          }
    }
    edit={[Function]}
    newComment={[Function]}
    remove={[Function]}
    showComment={[Function]}
    voteDown={[Function]}
    voteUp={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="btn btn-outline-dark btn-sm mr-sm-2"
          id="show"
          onClick={[Function]}
          type="button"
>
          Visualisar Comentários
           
          <span
                    className="badge badge-light"
          >
                    2
          </span>
</button>,
        <button
          className="btn btn-outline-dark btn-sm mr-sm-2"
          id="edit"
          onClick={[Function]}
          type="button"
>
          Adicionar Comentário
</button>,
        <button
          className="btn btn-outline-dark btn-sm mr-sm-2"
          id="edit"
          onClick={[Function]}
          type="button"
>
          Editar
</button>,
        <button
          className="btn btn-outline-dark btn-sm mr-sm-2"
          id="delete"
          onClick={[Function]}
          type="button"
>
          Excluir
</button>,
        <button
          className="btn btn-outline-success btn-sm mr-sm-2"
          id="like"
          onClick={[Function]}
          type="button"
>
          Gostei
</button>,
        <button
          className="btn btn-outline-danger btn-sm"
          id="dislike"
          onClick={[Function]}
          type="button"
>
          Não Gostei
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Visualisar Comentários",
            " ",
            <span
              className="badge badge-light"
>
              2
</span>,
          ],
          "className": "btn btn-outline-dark btn-sm mr-sm-2",
          "id": "show",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": Array [
          "Visualisar Comentários",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 2,
              "className": "badge badge-light",
            },
            "ref": null,
            "rendered": 2,
            "type": "span",
          },
        ],
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Adicionar Comentário",
          "className": "btn btn-outline-dark btn-sm mr-sm-2",
          "id": "edit",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Adicionar Comentário",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Editar",
          "className": "btn btn-outline-dark btn-sm mr-sm-2",
          "id": "edit",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Editar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Excluir",
          "className": "btn btn-outline-dark btn-sm mr-sm-2",
          "id": "delete",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Excluir",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Gostei",
          "className": "btn btn-outline-success btn-sm mr-sm-2",
          "id": "like",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Gostei",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Não Gostei",
          "className": "btn btn-outline-danger btn-sm",
          "id": "dislike",
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Não Gostei",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="btn btn-outline-dark btn-sm mr-sm-2"
            id="show"
            onClick={[Function]}
            type="button"
>
            Visualisar Comentários
             
            <span
                        className="badge badge-light"
            >
                        2
            </span>
</button>,
          <button
            className="btn btn-outline-dark btn-sm mr-sm-2"
            id="edit"
            onClick={[Function]}
            type="button"
>
            Adicionar Comentário
</button>,
          <button
            className="btn btn-outline-dark btn-sm mr-sm-2"
            id="edit"
            onClick={[Function]}
            type="button"
>
            Editar
</button>,
          <button
            className="btn btn-outline-dark btn-sm mr-sm-2"
            id="delete"
            onClick={[Function]}
            type="button"
>
            Excluir
</button>,
          <button
            className="btn btn-outline-success btn-sm mr-sm-2"
            id="like"
            onClick={[Function]}
            type="button"
>
            Gostei
</button>,
          <button
            className="btn btn-outline-danger btn-sm"
            id="dislike"
            onClick={[Function]}
            type="button"
>
            Não Gostei
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Visualisar Comentários",
              " ",
              <span
                className="badge badge-light"
>
                2
</span>,
            ],
            "className": "btn btn-outline-dark btn-sm mr-sm-2",
            "id": "show",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": Array [
            "Visualisar Comentários",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 2,
                "className": "badge badge-light",
              },
              "ref": null,
              "rendered": 2,
              "type": "span",
            },
          ],
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Adicionar Comentário",
            "className": "btn btn-outline-dark btn-sm mr-sm-2",
            "id": "edit",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Adicionar Comentário",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Editar",
            "className": "btn btn-outline-dark btn-sm mr-sm-2",
            "id": "edit",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Editar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Excluir",
            "className": "btn btn-outline-dark btn-sm mr-sm-2",
            "id": "delete",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Excluir",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Gostei",
            "className": "btn btn-outline-success btn-sm mr-sm-2",
            "id": "like",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Gostei",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Não Gostei",
            "className": "btn btn-outline-danger btn-sm",
            "id": "dislike",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Não Gostei",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
